galley/pkg/config/source/kube/apiserver/watcher.go

```diff
@@ -107,17 +107,17 @@ func (w *watcher) dispatch(h event.Handler) {
 }
 
 func (w *watcher) handleEvent(c event.Kind, obj interface{}) {
-	object, ok := obj.(metav1.Object)
-	if !ok {
-		if obj = tombstone.RecoverResource(obj); object != nil {
+	if _, ok := obj.(metav1.Object); !ok {
+		recoveredObject := tombstone.RecoverResource(obj)
```

Comment:
nit: If you use `obj` in this assignment instead of `recoveredObject` you could save some lines by having the comparison in the same `if` statement and not needing the assignment `obj = recoveredObject` below.

Reply:
I actually purposefully wrote it the longer way, because I found the original code confusing and I wanted to avoid possible future issues with shadowing vars by initializing them inside the conditional.